### PR TITLE
WIP: some refactoring of event handling w/ celticminstrel

### PR DESCRIPTION
**Pull request by Vultraz**
Refactored some code to use std::list instead of std::vector to avoid problems where the vector could be modified externally.

### CODE REVIEW

File: src/events.cpp [hunk 1]
```diff
@@ -66,52 +55,60 @@ bool context::remove_handler(sdl_handler* ptr)
 	static int depth = 0;
 	++depth;
 
-	//the handler is most likely on the back of the events array,
-	//so look there first, otherwise do a complete search.
+	// The handler is most likely on the back of the events list,
+	// so look there first, otherwise do a complete search.
 	if(handlers.back() == ptr) {
-		delete_handler_index(handlers.size()-1);
+		if(*focused_handler == ptr) {
```
Comment (CelticMinstrel) [inline src/events.cpp]: Hmm, thinking about it, this could actually cause a problem if there's only one element...?

Comment (Vultraz) [inline src/events.cpp]: Do we want to add the ternary check from below up here?

Comment (CelticMinstrel) [inline src/events.cpp]: You could. If there's only one element, you'd set it to end(), otherwise decrement it.

**Comment by Vultraz** [comment]
Closing this, since there's an expanded version in PR #696.

**Pull request closed without merging**